src/main.rs

```diff
@@ -146,9 +150,14 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
     // TODO: load from config
     let monitored = vec![devenv_deposit_address(&args.signers_xonly)];
 
-    let deposit_monitor = DepositMonitor::new(context.clone(), monitored);
+    let mut deposit_monitor = DepositMonitor::new(context.clone(), monitored);
 
-    runloop(context.clone(), &deposit_monitor, config.polling_interval).await;
+    runloop(
+        context.clone(),
```

Comment:
nit: I don't think that you need to clone the context here

Reply:
true. [249798d](https://github.com/BitcoinL2-Labs/sPoX/pull/8/commits/249798d6fa5fc245493a5ab990e8f69ed9a8673e)